core/src/test/scala/integration/kafka/api/MetricsTest.scala

```diff
@@ -44,7 +45,7 @@ class MetricsTest extends IntegrationTestHarness with SaslSetup {
   this.serverConfig.setProperty(KafkaConfig.ZkEnableSecureAclsProp, "false")
   this.serverConfig.setProperty(KafkaConfig.AutoCreateTopicsEnableDoc, "false")
   this.producerConfig.setProperty(ProducerConfig.LINGER_MS_CONFIG, "10")
-  this.producerConfig.setProperty(ProducerConfig.BATCH_SIZE_CONFIG, "1000000")
+  this.producerConfig.setProperty(ProducerConfig.BATCH_SIZE_CONFIG, producerBufferSize.toString)
```

Comment:
A way to ensure your batch is large is to use large records. Then you don't need this at all.

Reply:
True, I wrongly thought this was the maximum batch size.